### PR TITLE
Fix some rendering glitches

### PR DESCRIPTION
**Pull request by BlockMen**
- Fixes particle draworder
- Fixes nodehighlighting

### CODE REVIEW

**Comment by SmallJoker** [comment]
Code looks good :+1:

**Comment by Zeno-** [comment]
How did moving the code fragment fix the problem?

(Sorry, I don't have time to look at the entire 1600 lines of the function right now :))

**Comment by BlockMen** [comment]
@Zeno- the alpha texture sorting is kinda weird, so the last "special" element of generation special mesh function is drawn last. So if the highlightbox mesh is not added as last element the texture are not visible, since the information for blending is just missing.

**Comment by Zeno-** [comment]
Can you add a comment that says something like that in the source code?

File: src/content_mapblock.cpp [hunk 1]
```diff
@@ -1766,5 +1718,53 @@ void mapblock_mesh_generate_special(MeshMakeData *data,
...
+			std::vector<aabb3f> boxes = n.getSelectionBoxes(nodedef);
+			TileSpec h_tile;			
+			h_tile.material_flags |= MATERIAL_FLAG_HIGHLIGHTED;
+			h_tile.texture = tsrc->getTexture("halo.png",&h_tile.texture_id);
+			v3f pos = intToFloat(p, BS);
+			f32 d = 0.05 * BS;
+			for(std::vector<aabb3f>::iterator
+					i = boxes.begin();
+					i != boxes.end(); i++)
+			{
```
Comment (nerzhul) [inline src/content_mapblock.cpp]: Can you fix this brace ?

**Comment by Zeno-** [comment]
@BlockMen can you add the comment and fix the brace so that this can be merged?

**Comment by BlockMen** [comment]
@Zeno- will do tomorrow

**Pull request merged**